lib/src/main/java/ca/jbrains/pos/PointOfSale.java

```diff
@@ -20,15 +42,16 @@ private static void displayToConsole(String message) {
         return new BufferedReader(simulateInputFromStdin).lines();
     }
 
-    public static String displaySellOneItem(String barcode, SaleController saleController) {
-        if ("".equals(barcode)) {
-            return "Scanning error: empty barcode";
-        }
-
-        String price = saleController.getPrice(barcode);
-        if (price != null)
-            return price;
+    public static String displaySellOneItem(BarcodeScannedHandler barcodeScannedHandler, Barcode barcode) {
+        Either<ProductNotFound, ProductFound> returnProduct = barcodeScannedHandler.handleScannedBarcode(barcode);
+        if (returnProduct.isRight())
```

Comment:
One more step: look at the methods in `Either` for either `bimap()` or `fold()`.

Reply:
nice, done.